lonestar/eda/cpu/aig-rewriting/algorithms/CutPool.cpp

```diff
@@ -57,8 +57,8 @@ CutPool::~CutPool() {
 
 inline void CutPool::alloc() {
 
-  this->entriesFree =
-      (char*)malloc((long int)(this->entrySize * this->blockSize));
+  this->entriesFree = (char*)malloc(
+      (long int)(this->entrySize * this->blockSize * sizeof(char*)));
```

Comment:
The pointer is of type `char*`, so the allocated space should be using `sizeof(char)`?

Reply:
Also `sizeof(char)` is always 1

Reply:
Here's how I understand it, FWIW. Even though the result of `malloc` is getting cast to `char*` here, it's actually later used as a `char**`. That intent wasn't reflected in the `malloc` size or in the computed offset from the base pointer for the buffer. Prior to this there were misaligned writes for type `char*` happening. I think it likely was also overriding portions of previously stored pointers due to these misaligned accesses. I'm surprised the previous version even worked. Unfortunately I know little about the purpose of this bit of code so I'm not entirely certain my fix is right either.

Reply:
I don't see space for the pointer included in https://github.com/IntelligentSoftwareSystems/Galois/pull/226/files/a6b504f7033ef24e99a086f7c98c8f80d2940084#diff-3c61213a4fd449ea9fd96523fd5b2eb2R41-R46, but then I'm not 100% sure what each of those terms mean.

Reply:
Whoops. Last comment was meant to be on the other thread. I'll copy it to there.